src/enzyme_ad/jax/Dialect/EnzymeXLAOps.td

```diff
@@ -1058,4 +1058,141 @@ def AffineStoreVar : EnzymeXLA_Op<"store_var", [Pure]> {
   let summary = "Fake store an SSA value for conversion to ISL";
 }
 
+// MPI Ops
+
+def MPICommRankOp : EnzymeXLA_Op<"comm_rank", []> {
+  let summary = "Equivalent to " "`MPI_Comm_rank(MPI_COMM_WORLD, &rank)`";
+
+  let arguments = (
+    ins AnyTensor : $inrank
```

Comment:
yeah this shouldn't require an inrank, it can just return the rank, same with size